File: jdk/Step04/src/singletone/Number.java
package singletone;

public class Number {
	private static Number instance = new Number();
	
	private int num;
	
	//private 생성자는 외부에서 생성이 안됨
	private Number() {	}
	
	public static Number getInstance() {
		if(instance == null)
			instance = new Number();
		return instance;
	}
	
	public void addNum() {
		num++;
	}
	
	public int getNum() {
		return num;
	}
	
}

File: C#/Step04/Step04/Number.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Step04
{
    public class Number
    {
        private static Number instance = new Number();
        private int num;
        private Number()
        {
            num = 0;
        }

        public static Number GetInstance()
        {
            instance.num++;
            if(instance == null)
                instance = new Number();
            return instance;
        }
        public int GetNum() { return num; }
    }
}
